blog/_src/posts/2016-07-25-tutorial-zero-to-sixty-in-racket.scrbl

```diff
@@ -0,0 +1,698 @@
+
+Racket has excellent tools for evolving scripts into stable and useful programs.
+This post demonstrates one such evolution and highlights the aspects 
+ of Racket that make for a smooth development process.
+
+<!-- more -->
+
+Why should anyone use @hyperlink["http://racket-lang.org/"]{Racket}?
+There are two reasons:
+@itemlist[#:style 'ordered
+  @item{
+    You have a problem that can only be solved with Racket's language-building tools
+     @; e.g., Rosette, Pollen
+  }
+  @item{
+    Racket is a nice language to program in.
+    (Has lexical scope, parentheses, a nice community...)
+  }
+]
+
+My favorite part of Racket is how it supports a certain development style of
+ evolving scripts into programs.
+When I start coding (after design, before debugging), I can focus the problem at hand.
+Next come examples, unit tests, and types to be sure the solution is correct.
+Finally, I worry about aesthetics, efficiency, and how the solution can be used as a library in a larger context.
+
+Bottom line: Racket lets me add features in order of importance and keep a
+ runnable program at all times.
+
+
+@section{Problem: A KWIC Index Production System}
+
+A KWIC index system
+@;@itemlist[
+@;  @item{
+    reads input from a file,
+@;  }
+@;  @item{
+    divides each line of the file into whitespace-separated words,
+@;  }
+@;  @item{
+    and outputs (in alphabetical order) all circular shifts of all lines.
+@;  }
+@;]
+
+The first circular shift of a line @racket{A B C} is the line @racket{B C A}.
+The second circular shift is @racket{C A B}.
+
+Building a KWIK index is an interesting historical problem.
+According to @hyperlink["https://www.cs.umd.edu/class/spring2003/cmsc838p/Design/criteria.pdf"]{D.L. Parnas (1972)}:
+
+@nested{Except under extreme circumstances (huge data base, no supporting software)
+  such a system could be implemented by a good programmer within a week or two.
+}
+
+The year is 2016, and I bet only @hyperlink["http://wiki.portal.chalmers.se/agda/pmwiki.php"]{Agda} and @hyperlink["https://scratch.mit.edu/"]{Scratch}
+ programmers would need the full two weeks.
+We'll be done in 20 minutes.
+
+
+@section{A Script}
+
+To start, open a file and type:
+
+@codeblock{
+  #lang racket
+}
+
+You can name the file anything, like @tt{kwik.rkt} or @tt{rkt.kwik} or @tt{foo}.
+Racket doesn't care, but it does need the @tt{#lang} line to read the
+ contents of the file.
+
+The first part of the solution is a function to read input from a file into a
+ list of strings for further processing.
+The built-in function @hyperlink["http://docs.racket-lang.org/reference/Filesystem.html#%28def._%28%28lib._racket%2Ffile..rkt%29._file-~3elines%29%29"]{file->lines} does exactly this, but we'll
+ for-loop instead.
+
+@(begin
+  #reader scribble/comment-reader
+  (racketblock
+    (define (kwik-read filename)
+      (with-input-from-file filename
+        (λ ()
+          (for/list ([line (in-lines)])
```

Comment:
No experience. At first I tried explaining for loops and `in-lines` here, but it got too long. So I decided to just leave it as "this is code, the code does X" and let the reader put together the "how?".